src/azure.ad/oauth2_outbound_handler.bal

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/oauth2;
+
+# Creates an outbound auth handler which can be used by an http client.
+# 
+# + oauth2Config - OAuth2 configuration for the client.
+# + return - Outbound auth handler.
+public function getAzureAdOutboundOAuth2BearerHandler(ClientCredentialsGrantConfig|PasswordGrantConfig|DirectTokenConfig oauth2Config) returns http:BearerAuthHandler {
```

Comment:
Shall we rename the function to `getOutboundOAuth2BearerHandler`. AzureAd is redundant since it is in the module prefix

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerinax-azure.ad/pull/1/commits/a534cb1f06c82d058707850878f95cd1e405c41b